src/box/lua/init.c

```diff
@@ -545,14 +545,14 @@ lbox_txn_iterator_next(struct lua_State *L)
 {
 	struct txn *txn = in_txn();
 	int64_t txn_id = luaL_toint64(L, lua_upvalueindex(1));
-	if (txn == NULL || txn->id != txn_id) {
-		diag_set(ClientError, ER_CURSOR_NO_TRANSACTION);
-		return luaT_error(L);
-	}
 	struct txn_stmt *stmt =
 		(struct txn_stmt *) lua_topointer(L, lua_upvalueindex(2));
 	if (stmt == NULL)
 		return 0;
+	if (txn == NULL || txn->id != txn_id) {
+		diag_set(ClientError, ER_CURSOR_NO_TRANSACTION);
+		return luaT_error(L);
+	}
```

Comment:
Why did you move this check? Looks like an unrelated change.

Reply:
Left over from previous version of the patch, fixed.